gnucash/report/reports/standard/investment-lots.scm

```diff
@@ -0,0 +1,2108 @@
+             (short-term-sold-value (get-report-value-zero))
+             (short-term-realized-gain (get-report-value-zero))
+             (long-term-sold-amount (get-amount-zero))
+             (long-term-sold-basis (get-report-value-zero))
+             (long-term-sold-value (get-report-value-zero))
+             (long-term-realized-gain (get-report-value-zero))
+             (end-amount (get-amount-zero))
+             (end-basis (get-report-value-zero))
+             (end-value (get-report-value-zero))
+             (unrealized-gain (get-report-value-zero))
+             (short-term-unrealized-gain (get-report-value-zero))
+             (long-term-unrealized-gain (get-report-value-zero))          
+             (has-warnings #f)
+             (is-active-in-window #f)
+             (currency '())
+
+             ;; private:
+             (splits-realized-gain (get-report-value-zero))
+             (splits-table (gnc:make-html-table))
+             (first-negative-split '())
+             (earliest-bought-split '())
+             (earliest-bought-split-date '())
+             (latest-bought-split-date '())
+             (last-sold-split '())
+
+             ;; For lot collectors only:
+             (lot '())
+
+             ;; For account collectors only:
+             (account '())
+             (unassigned-splits '()))
+
+        ;; Returns whether this instance pertains to a single lot.
+        (define (get-is-lot-stats)
+          (not (null? lot)))
+
+        ;; Returns whether this instance pertains to all lots within a
+        ;; single account.
+        (define (get-is-account-stats)
+          (and (not (get-is-lot-stats))
+               (not (null? account))))
+
+        ;; Returns whether this instance pertains to all lots within all
+        ;; accounts (i.e. the grand total).
+        (define (get-is-grand-total-stats)
+          (not (or (get-is-lot-stats)
+                   (get-is-account-stats))))        
+
+        ;; Initializes the instance with an account's context.
+        (define (init-for-account accnt unassigned-splts)
+          (set! account accnt)
+          (set! unassigned-splits unassigned-splts))
+
+        ;; Adds to the given table a row for the given split.
+        (define (add-split-row
+                  split
+                  trans-date
+                  bought-amount
+                  bought-value
+                  sold-amount
+                  sold-basis
+                  sold-value
+                  sold-gain
+                  is-long-term)
+          (if show-split-rows
+            (let ((date-cell
+                      (to-split-cell (qof-print-date trans-date) split)))
+              (add-data-row
+                splits-table
+                currency
+                #f ;; is-bold
+                #t ;; is-odd-row
+                (string-append
+                    ;; indentation
+                    "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"
```

Comment:
There's 1 remaining nbsp in html-acct-table.scm and is proving difficult to reomve. We want to avoid any further raw html in code. It should be replaced by a `bool` to indicate whether to use an additional column, or merge 2 columns.

Reply:
I replaced the nbsp's with text-indent css style.